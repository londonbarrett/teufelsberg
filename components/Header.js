import * as React from 'react';
import PropTypes from 'prop-types';

const Header = ({ title }) => (
  <header className="header">
    <h1 className="title">{title}</h1>
    <style jsx>
      {`
        .header {
          background: black;
          color: white;
          overflow: auto;
          width: 100%;
        }
        .title {
          font-size: 2rem;
          line-height: 3rem;
          margin: 1rem;
        }
      `}
    </style>
  </header>
);

Header.propTypes = {
  title: PropTypes.string.isRequired,
};

export default Header;
