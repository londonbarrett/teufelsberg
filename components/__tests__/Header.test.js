/* eslint-env jest */

import React from 'react';
import renderer from 'react-test-renderer';
import Header from '../Header';

describe('Header Tests', () => {
  it('Renders correctly', () => {
    const component = renderer.create(<Header title="Header title" />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
