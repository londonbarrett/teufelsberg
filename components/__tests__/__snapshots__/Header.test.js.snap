// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Tests Renders correctly 1`] = `
<header
  className="header"
>
  <h1
    className="title"
  >
    Header title
  </h1>
</header>
`;
